packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -1,75 +1,135 @@
+import useThemeProps from '../styles/useThemeProps';
 import InputBase from '../InputBase';
 import MenuItem from '../MenuItem';
 import Select from '../Select';
 import TableCell from '../TableCell';
 import Toolbar from '../Toolbar';
-import Typography from '../Typography';
 import TablePaginationActions from './TablePaginationActions';
 import useId from '../utils/useId';
+import tablePaginationClasses, { getTablePaginationUtilityClass } from './tablePaginationClasses';
+
+const TablePaginationRoot = experimentalStyled(
+  TableCell,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)(({ theme }) => ({
+  overflow: 'auto',
+  color: theme.palette.text.primary,
+  fontSize: theme.typography.pxToRem(14),
+  // Increase the specificity to override TableCell.
+  '&:last-child': {
+    padding: 0,
+  },
+}));
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    color: theme.palette.text.primary,
-    fontSize: theme.typography.pxToRem(14),
-    overflow: 'auto',
-    // Increase the specificity to override TableCell.
-    '&:last-child': {
-      padding: 0,
-    },
+const TablePaginationToolbar = experimentalStyled(
+  Toolbar,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Toolbar',
+    overridesResolver: (props, styles) => styles.toolbar,
   },
-  /* Styles applied to the Toolbar component. */
-  toolbar: {
+)(({ theme }) => ({
+  minHeight: 52,
+  paddingRight: 2,
+  [`${theme.breakpoints.up('xs')} and (orientation: landscape)`]: {
     minHeight: 52,
-    paddingRight: 2,
   },
-  /* Styles applied to the spacer element. */
-  spacer: {
-    flex: '1 1 100%',
+  [theme.breakpoints.up('sm')]: {
+    minHeight: 52,
+    paddingRight: 2,
   },
-  /* Styles applied to the select label Typography element. */
-  selectLabel: {
+  [`& .${tablePaginationClasses.actions}`]: {
     flexShrink: 0,
+    marginLeft: 20,
+  },
+}));
+
+const TablePaginationSpacer = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Spacer',
+    overridesResolver: (props, styles) => styles.spacer,
   },
-  // TODO v5: `.selectRoot` should be merged with `.input`
-  /* Styles applied to the Select component root element. */
-  selectRoot: {
-    marginRight: 32,
-    marginLeft: 8,
+)({
+  flex: '1 1 100%',
+});
+
+const TablePaginationSelectLabel = experimentalStyled(
+  'p',
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'SelectLabel',
+    overridesResolver: (props, styles) => styles.selectLabel,
+  },
+)(({ theme }) => ({
+  ...theme.typography.body2,
+  flexShrink: 0,
+}));
+
+const TablePaginationSelect = experimentalStyled(
+  Select,
+  {},
+  {
+    name: 'MuiTablePagination',
+    slot: 'Select',
+    overridesResolver: (props, styles) => ({
+      [`& .${tablePaginationClasses.selectIcon}`]: styles.selectIcon,
+      [`& .${tablePaginationClasses.select}`]: styles.select,
```

Comment:
Why is there a selector here? Shouldn't this be simply:
```suggestion
      ...styles.select,
```

Reply:
Cannot, because the `select` slot in the children of the styled component